### PR TITLE
Decomposition of BecomeACopyOf

### PR DESCRIPTION
**Pull request by MichaelEbert**
Changed BecomeACopyOf() into Become() and ACopyOf().
Requires wordbots-parser/pull/67 at the same time.

### CODE REVIEW

File: src/common/constants.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const DISABLE_AI = false;
 export const DISPLAY_HEX_IDS = false;
 export const ENABLE_REDUX_TIME_TRAVEL = false;
 const USE_LOCAL_PARSER = false;
+const LOCAL_PARSER_PORT = 8080;
```
Comment (AlexNisnevich) [inline src/common/constants.js]: I guess you should change this to match the port you're setting in `wordbots-parser`.

File: src/common/vocabulary/targets.js [hunk 2]
```diff
@@ -155,6 +154,12 @@ export default function targets(state, currentObject) {
 
     thisRobot: function () {
       return {type: 'objects', entries: [currentObject]};
+    },
+
+    //a generic 1/1/1 robot.
```
Comment (AlexNisnevich) [inline src/common/vocabulary/targets.js]: Write a TODO that this will be more customizable later.

File: src/common/vocabulary/targets.js [hunk 3]
```diff
@@ -155,6 +154,12 @@ export default function targets(state, currentObject) {
...
+    token: function (){
+      return {abilities:[],baseCost:0,cost:0,id: 'builtin/token',name:'Token',source:'builtin',stats:{attack:1,health:1,speed:1},type:0};
```
Comment (AlexNisnevich) [inline src/common/vocabulary/targets.js]: Maybe this token should be defined in `store/cards.js` for now.

File: src/common/vocabulary/actions.js [hunk 4]
```diff
@@ -20,6 +21,31 @@ export default function actions(state) {
...
+        });
+      });
+
+      //set triggers
+      if (card.abilities && card.abilities.length > 0) {
+        card.abilities.forEach((cmd, idx) => {
+          const cmdText = splitSentences(card.text)[idx];
+          state.currentCmdText = cmdText.includes('"') ? cmdText.split('"')[1].replace(/"/g, '') : cmdText;
+
+          iterateOver(sources)(source=>{executeCmd(state, cmd, source);});//works for multitarget? who knows!
```
Comment (AlexNisnevich) [inline src/common/vocabulary/actions.js]: So does it work?

File: src/common/vocabulary/actions.js [hunk 5]
```diff
@@ -20,6 +21,31 @@ export default function actions(state) {
   };
 
   return {
+    //become a card(becomeACopy takes a playedObject, this takes a card)
```
Comment (AlexNisnevich) [inline src/common/vocabulary/actions.js]: nit (here and below)- Our convention for comments is space before and after the `//` and sentence case.

File: src/common/vocabulary/actions.js [hunk 6]
```diff
@@ -20,6 +21,31 @@ export default function actions(state) {
...
+    become: function (sources, card) {
+      //console.log(card);
```
Comment (AlexNisnevich) [inline src/common/vocabulary/actions.js]: nit- You can remove the commented-out line

File: src/common/vocabulary/actions.js (hunk 4 above)
Comment (MichaelEbert) [reply src/common/vocabulary/actions.js]: i think so? I still don't completely understand the client, so the code might be broken in some hidden way somewhere, but I don't know what to look/test for.

**Comment by MichaelEbert** [comment]
Most changes fixed.
Moved changes to creation-effects branch to mirror parser.
In the future, I think i'll do that (create branches with the same name) whenever I need to implement matching changes for both parser and core.

>Could you write a test for this in test/reducers/game.spec.js just so we can be sure it works? (This will 
>require adding a card to test/data/cards.js.)
It would also require figuring out how the client works. Besides, it doesn't BREAK any test cases...

**Pull request closed without merging**